### DISCUSSION TITLE
Enable ingress and Cors via. apphost or environment variables

### DISCUSSION THREAD

**Discussion by SalomonHenke** [microsoft/aspire#4379] [Q&A] [answered]
Hello All 
At the fear of sounding like a complete newbie, i just want to clarify - i am. 

I have deployed my containers via azd up and later updates via. azd deploy. However 
- Cors is defaulting to No policy (where im looking for an allow All (*))  - The app itself has a Cors policy of allow-all, but i suppose azd / the genereated manifest doesnt account for the apps own policy and just creates an image. 
- Ingress is defaulting to internal only where im looking for External. 
Is there a way for me to set this beforehand? Is this supposed to be an AZD job to not pollute aspire with vendor-specific options? 

Also, if i change it and redeploy it resets. Which, is this because it takes down the existing containers and redeploys them from the manifest (which im pressuming defaults to ingress interal and no cors policies)?

Thanks in advance. It really helps a new software developer understand these things.

**Comment by davidfowl** [comment, marked as answer]
> Ingress is defaulting to internal only where im looking for External.
Is there a way for me to set this beforehand? Is this supposed to be an AZD job to not pollute aspire with vendor-specific options?

[WithExternalHttpEndpoints](https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.resourcebuilderextensions.withexternalhttpendpoints?view=dotnet-aspire-8.0) on your resource will mark all http or https endpoints as external. Endpoints in the appmodel have an "external" boolean and this will get translated into making the generated infra for the container app external as well.

> Cors is defaulting to No policy (where im looking for an allow All (*)) - The app itself has a Cors policy of allow-all, but i suppose azd / the genereated manifest doesnt account for the apps own policy and just creates an image.

With respect to changing container apps properties itself, this is the current answer:

https://github.com/dotnet/aspire/discussions/4337#discussioncomment-9634038

> Also, if i change it and redeploy it resets. Which, is this because it takes down the existing containers and redeploys them from the manifest (which im pressuming defaults to ingress interal and no cors policies)?

You can see the default container app policy using `azd infra synth`, we set single revision mode by default https://learn.microsoft.com/en-us/azure/container-apps/revisions#revision-modes.

I recommend reading the in-depth deployment docs as a way to explore more about what's happening https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/aca-deployment-azd-in-depth?tabs=macos. Feel free to ask more questions as you learn more!

**Reply by SalomonHenke** [reply]
Thanks alot. This makes sense. I promise i did read the documentation  😂 

I got to this part: 
![image](https://github.com/dotnet/aspire/assets/96744651/b0670583-ff1f-4871-acfb-019e09edbe62)

Which i thought was why something had gone wrong, since i didn't see it in my project, which i now assume is cause i had none of my clients marked as public? So it never gneerated. Anyways thanks alot. Aspire is kickass to use.

**Reply by davidfowl** [reply]
@vhvb1989 can we update this doc?

**Reply by vhvb1989** [reply]
https://github.com/dotnet/docs-aspire/pull/1052
